homeassistant/components/kaleidescape/entity.py

```diff
@@ -0,0 +1,69 @@
+"""Base Entity for Kaleidescape."""
+
+from __future__ import annotations
+
+import logging
+from typing import TYPE_CHECKING
+
+from kaleidescape import const as kaleidescape_const
+
+from homeassistant.const import STATE_IDLE, STATE_OFF, STATE_PAUSED, STATE_PLAYING
+from homeassistant.core import callback
+from homeassistant.helpers.entity import DeviceInfo, Entity
+
+from .const import DOMAIN as KALEIDESCAPE_DOMAIN, NAME as KALEIDESCAPE_NAME
+
+if TYPE_CHECKING:
+    from kaleidescape import Device as KaleidescapeDevice
+
+KALEIDESCAPE_PLAYING_STATES = [
+    kaleidescape_const.PLAY_STATUS_PLAYING,
+    kaleidescape_const.PLAY_STATUS_FORWARD,
+    kaleidescape_const.PLAY_STATUS_REVERSE,
+]
+
+KALEIDESCAPE_PAUSED_STATES = [kaleidescape_const.PLAY_STATUS_PAUSED]
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class KaleidescapeEntity(Entity):
+    """Defines a base Kaleidescape entity."""
+
+    def __init__(self, device: KaleidescapeDevice) -> None:
+        """Initialize entity."""
+        self._device = device
+
+        self._attr_should_poll = False
+        self._attr_unique_id = device.serial_number
+        self._attr_name = f"{KALEIDESCAPE_NAME} {device.system.friendly_name}"
+        self._attr_device_info = DeviceInfo(
+            identifiers={(KALEIDESCAPE_DOMAIN, self._device.serial_number)},
```

Comment:
If the mac address is available, its generally preferred to set `connections` over `identifiers`

Reply:
Unfortunately the Kaleidescape API does not expose the mac address.